test/shop/specs/features/01-withdraw-money.api.steps.ts

```diff
@@ -0,0 +1,115 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { defineFeature, loadFeature } from 'jest-cucumber';
+import { INestApplication } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { TransactionModule } from '../../../../src/transaction/transaction.module';
+
+import * as supertest from 'supertest';
+import PersonEntity from '../../../../src/database/entities/person.entity';
+import WalletEntity from '../../../../src/database/entities/wallet.entity';
+import ProductEntity from '../../../../src/database/entities/product.entity';
+import PersonProductEntity from '../../../../src/database/entities/person-products.entity';
+import setupSqliteModule from './utils/db/sqlite';
+
+const feature = loadFeature(
+  'test/shop/specs/features/01-withdraw-money.feature',
+);
+
+defineFeature(feature, (test) => {
+  let app: INestApplication;
+  let agent;
+  let personRepository: Repository<PersonEntity>;
+  let walletRepository: Repository<WalletEntity>;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TransactionModule,
+        setupSqliteModule([
+          PersonEntity,
+          WalletEntity,
+          ProductEntity,
+          PersonProductEntity,
+        ]),
+      ],
+    }).compile();
+
+    app = await module.createNestApplication();
+
+    agent = supertest.agent(app.getHttpServer());
+    await app.init();
+
+    personRepository = module.get('PersonEntityRepository');
+    walletRepository = module.get('WalletEntityRepository');
+  });
+
+  test('Withdraw money from the wallet', ({ given, when, then, and }) => {
+    let response;
+    let person: PersonEntity;
+
+    afterAll(async () => {
+      await personRepository.delete(person.id);
+      await walletRepository.delete(person.wallet.id);
+    });
+
+    given('There is a wallet with $5', async () => {
+      person = await personRepository.save({
+        name: 'Test',
+        wallet: {
+          balance: 5,
+        },
+        cash: 0,
+      });
+    });
+
+    when('I withdraw the money $2 from the wallet', async () => {
+      response = await agent
+        .post(`/transactions/persons/${person.id}/withdraw-money`)
+        .send({ amount: 2 });
+    });
+
+    then('I have $2 in money', async () => {
+      expect(response.body.cash).toBe(2);
+    });
+
```

Comment:
I'm not sure if there is a need to keep a cache in the database. I think this check is all that is needed in the context of the cache.